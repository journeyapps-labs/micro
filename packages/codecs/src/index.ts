export * from './parsers';
export * from './codecs';
