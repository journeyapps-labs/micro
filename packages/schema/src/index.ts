export * from './json-schema/keywords';
export * from './json-schema/parser';

export * from './definitions';
export * from './validators';
export * from './utils';
