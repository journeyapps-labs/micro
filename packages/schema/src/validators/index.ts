export * from './ts-codec-validator';
export * from './schema-validator';
export * from './zod-validator';
