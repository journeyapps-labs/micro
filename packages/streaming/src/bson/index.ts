export * from "./buffer-array";
export * from "./constants";
export * from "./encoder";
export * from "./decoder";
export * from "./header";
