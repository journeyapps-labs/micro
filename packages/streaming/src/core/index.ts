export * from './backpressure';
export * from './transformers';
export * from './node-utils';
export * from './utils';
