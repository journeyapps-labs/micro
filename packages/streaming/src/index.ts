export * as compat from './core/cross-stream';
export * as middleware from './middleware';
export * as bson from './bson';
export * from './core';
