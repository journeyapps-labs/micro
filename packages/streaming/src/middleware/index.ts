export * from './streamed-body-parser-v1';
export * from './streamed-body-parser-v2';
