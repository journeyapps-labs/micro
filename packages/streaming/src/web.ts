export * as compat from "./core/cross-stream";
export * as bson from "./bson";

export * from "./core/transformers";
export * from "./core/utils";
