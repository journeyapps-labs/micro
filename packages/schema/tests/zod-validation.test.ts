import { describe, test, expect } from "vitest";

import * as micro_schema from "../src";
import * as t from "zod";

describe("zod-validation", () => {
  const schema = t.object({
    name: t.string(),
    surname: t.string(),
    other: t.object({
      a: t.array(t.string()),
      b: t.literal("optional").optional(),
    }),
  });

  test("passes validation for runtime codec", () => {
    const validator = micro_schema.createZodValidator(schema);

    const result = validator.validate({
      name: "a",
      surname: "b",
      other: {
        a: ["nice"],
        b: "optional",
      },
    });

    expect(result).toMatchSnapshot();
  });

  test("fails validation for runtime codec", () => {
    const validator = micro_schema.createZodValidator(schema);

    const result = validator.validate({
      // @ts-ignore
      name: 1,
      other: {
        // @ts-ignore
        b: "op",
      },
    });

    expect(result).toMatchSnapshot();
  });
});
