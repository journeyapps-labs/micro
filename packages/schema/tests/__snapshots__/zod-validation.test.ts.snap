// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`zod-validation > fails validation for runtime codec 1`] = `
{
  "errors": [
    "{"_errors":[],"name":{"_errors":["Expected string, received number"]},"surname":{"_errors":["Required"]},"other":{"_errors":[],"a":{"_errors":["Required"]},"b":{"_errors":["Invalid literal value, expected \\"optional\\""]}}}",
  ],
  "valid": false,
}
`;

exports[`zod-validation > passes validation for runtime codec 1`] = `
{
  "valid": true,
}
`;
